### DISCUSSION TITLE
How do you get block data?

### DISCUSSION THREAD

**Discussion by henrikwirth** [WebDevStudios/nextjs-wordpress-starter#272] [Q&A]
Hi guys, great starter.

Just diving through to understand your setup a little more since I'd like to move towards NextJS from Gatsby setups and also would like to use Gutenberg instead of ACF Flexible Layouts.

Now I'm wondering how you get the Gutenberg data as `block` JSON? 

I don't think this is part of standard WPGraphQL but I couldn't find anything related to it in your codebase. What am I missing :D ?

**Comment by reinis7** [comment]
You can see the getPostTypeProps() function in the getStaticProps() that fetch props from the WP GraphQL API.
That getPostTypeProps function would have several conditional apolloState. 
Every apolloState will be updated with response from condition matching request such like getPostById.

**Comment by gregrickaby** [comment]
@henrikwirth Thanks for the question. @reinis7 Answered correctly that we're grabbing `blocksJSON` via `getStaticProps()`.

The `blocksJSON` endpoint is made available via https://www.wpgraphql.com/extenstion-plugins/wpgraphql-gutenberg/

I wrote a blog post about this too: https://webdevstudios.com/2021/04/20/quick-tip-gutenberg-blocks-as-json/

Hope this helps!

**Reply by henrikwirth** [reply]
Hi @gregrickaby & @reinis7. Thanks for the answers. The missing piece for me was the use of Peters extension for Gutenberg. Was wondering if there was another way to solve this already.
